src/modules/usenet/UsenetQueueList.tsx

```diff
@@ -0,0 +1,159 @@
+import {
+  ActionIcon,
+  Alert,
+  Center,
+  Code,
+  Group,
+  Pagination,
+  Progress,
+  Skeleton,
+  Table,
+  Text,
+  Title,
+  Tooltip,
+  useMantineTheme,
+} from '@mantine/core';
+import { IconAlertCircle, IconPlayerPause, IconPlayerPlay } from '@tabler/icons';
+import { AxiosError } from 'axios';
+import dayjs from 'dayjs';
+import duration from 'dayjs/plugin/duration';
+import { useTranslation } from 'next-i18next';
+import { FunctionComponent, useState } from 'react';
+import { useGetUsenetDownloads } from '../../tools/hooks/api';
+import { humanFileSize } from '../../tools/humanFileSize';
+
+dayjs.extend(duration);
+
+interface UsenetQueueListProps {
+  serviceId: string;
+}
+
+const PAGE_SIZE = 10;
+
+export const UsenetQueueList: FunctionComponent<UsenetQueueListProps> = ({ serviceId }) => {
+  const theme = useMantineTheme();
+  const { t } = useTranslation('modules/usenet');
+
+  const [page, setPage] = useState(1);
+  const { data, isLoading, isError, error } = useGetUsenetDownloads({
+    limit: PAGE_SIZE,
+    offset: (page - 1) * PAGE_SIZE,
+    serviceId,
+  });
+  const totalPages = Math.ceil((data?.total || 1) / PAGE_SIZE);
+
+  if (isLoading) {
+    return (
+      <>
+        <Skeleton height={40} mt={10} />
+        <Skeleton height={40} mt={10} />
+        <Skeleton height={40} mt={10} />
+      </>
+    );
+  }
+
+  if (isError) {
+    return (
+      <Group position="center">
+        <Alert icon={<IconAlertCircle size={16} />} my="lg" title="Error!" color="red" radius="md">
+          {t('queue.error')}
+          <Code mt="sm" block>
+            {(error as AxiosError)?.response?.data as string}
+          </Code>
+        </Alert>
+      </Group>
+    );
+  }
+
+  if (!data || data.items.length <= 0) {
+    return (
+      <Center style={{ display: 'flex', alignItems: 'center', justifyContent: 'center' }}>
+        <Title order={3}>{t('queue.empty')}</Title>
+      </Center>
+    );
+  }
+
+  return (
+    <>
+      <Table highlightOnHover style={{ tableLayout: 'fixed' }}>
+        <thead>
+          <tr>
+            <th style={{ width: 50 }} />
+            <th style={{ width: '75%' }}>{t('queue.header.name')}</th>
+            <th style={{ width: 100 }}>{t('queue.header.size')}</th>
+            <th style={{ width: 100 }}>{t('queue.header.eta')}</th>
+            <th style={{ width: 200 }}>{t('queue.header.progress')}</th>
+          </tr>
+        </thead>
+        <tbody>
+          {data.items.map((nzb) => (
+            <tr key={nzb.id}>
+              <td>
+                {nzb.state === 'paused' ? (
+                  <Tooltip label="NOT IMPLEMENTED">
```

Comment:
Is this a TODO?

Reply:
Yeah, still need to implement this but the rest of this is completly usuable as is. I just didn't want user clicking the button and thinking it didn't work.

Reply:
Letting the user know what's going on is very good practice. Do you want to implement this later? If yes, you can resolve this conv.